src/main/java/frc/robot/subsystems/drivetrain/commands/AdjustToTargetOnCommand.java

```diff
@@ -0,0 +1,61 @@
+package frc.robot.subsystems.drivetrain.commands;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.Robot;
+import frc.robot.subsystems.drivetrain.SwerveDrive;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.DoubleSupplier;
+
+public class AdjustToTargetOnCommand extends CommandBase {
+    private final SwerveDrive swerveDrive;
+    private final DoubleSupplier yawSupplier;
+    private final PIDController adjustController = new PIDController(Constants.SwerveDrive.ADJUST_CONTROLLER_KP.get(), 0, 0) {{
+        enableContinuousInput(-Math.PI, Math.PI);
+        setTolerance(Constants.SwerveDrive.ADJUST_CONTROLLER_TOLERANCE);
+    }};
+    private final BooleanSupplier hasTarget;
+    private Rotation2d target;
+
+
+    public AdjustToTargetOnCommand(SwerveDrive swerveDrive, DoubleSupplier yawSupplier, BooleanSupplier hasTarget) {
+        this.swerveDrive = swerveDrive;
+        this.yawSupplier = yawSupplier;
+        this.hasTarget = hasTarget;
+        target = Robot.getAngle();
```

Comment:
Needs to be initialized in `initialize` (you don't have the gyro at compile time)

Reply:
I do already initialize it in initialize. I could have just as well removed that line at 28. it's just for safety.